meadow-client/src/Semantics.purs

```diff
@@ -0,0 +1,1067 @@
+  show = showObservation
+
+data Contract =
+    Null |
+    Commit IdAction IdCommit Person Value Timeout Timeout Contract Contract |
+    Pay IdAction IdCommit Person Value Timeout Contract Contract |
+    Both Contract Contract |
+    Choice Observation Contract Contract |
+    When Observation Timeout Contract Contract |
+    While Observation Timeout Contract Contract |
+    Scale Value Value Value Contract |
+    Let LetLabel Contract Contract |
+    Use LetLabel
+derive instance eqContract :: Eq Contract  
+derive instance ordContract :: Ord Contract  
+
+showContract :: Contract -> String
+showContract Null = "Null"
+showContract (Commit idAction idCommit person value timeout1 timeout2 contract1 contract2) =
+  joinWith "" ["Commit ",
+               toString idAction, " ",
+               toString idCommit, " ",
+               toString person, " (",
+               showValue value, ") ",
+               toString timeout1, " ",
+               toString timeout2, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Pay idAction idCommit person value timeout contract1 contract2) =
+  joinWith "" ["Pay ",
+               toString idAction, " ",
+               toString idCommit, " ",
+               toString person, " (",
+               showValue value, ") ",
+               toString timeout, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Both contract1 contract2) =
+  joinWith "" ["Both (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Choice observation contract1 contract2) =
+  joinWith "" ["Choice (",
+               showObservation observation, ") (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (When observation timeout contract1 contract2) =
+  joinWith "" ["When (",
+               showObservation observation, ") ",
+               toString timeout, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (While observation timeout contract1 contract2) =
+  joinWith "" ["While (",
+               showObservation observation, ") ",
+               toString timeout, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Scale value1 value2 value3 contract) =
+  joinWith "" ["Scale (",
+               showValue value1, ") (",
+               showValue value2, ") (",
+               showValue value3, ") (",
+               showContract contract, ")"]
+showContract (Let letLabel contract1 contract2) =
+  joinWith "" ["Let ",
+               toString letLabel, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Use letLabel) =
+  joinWith "" ["Use ",
+               toString letLabel]
```

Comment:
I've got some pamph- No? Still busy huh? Okay, I'll call again tomorrow.

Reply:
Hm, I guess it wouldn't hurt to have a look.